crates/sui-storage/src/lib.rs

```diff
@@ -44,3 +45,14 @@ pub fn default_db_options(
 
     (options, point_lookup)
 }
+
+// Used to exec futures that send data to/from other threads. In the simulator, this effectively
+// becomes a blocking call, which removes the non-determinism that would otherwise be caused by the
+// timing of the reply from the other thread.
+pub(crate) async fn exec_client_future<F: Future>(fut: F) -> <F as Future>::Output {
+    if cfg!(msim) {
+        futures::executor::block_on(fut)
```

Comment:
I am sort of surprised this worked - normally you can't call block_on from inside tokio thread

Reply:
You can't call Runtime::block_on (I think that panics?) - but this is just a short-hand for calling poll() in a loop until it returns Ready. There's nothing to stop you from doing that.

Obviously this will deadlock if your future is waiting for something to happen on the current thread, but that's explicitly not the case here - we are sending a message to another thread and waiting for it to come back. So ultimately this boils down to just waiting on the condvar inside the channel, which is a perfectly reasonable thing to do.